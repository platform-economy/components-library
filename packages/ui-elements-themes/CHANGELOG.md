# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
