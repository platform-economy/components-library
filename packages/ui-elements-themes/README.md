# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
