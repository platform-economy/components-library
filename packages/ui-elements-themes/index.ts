export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
