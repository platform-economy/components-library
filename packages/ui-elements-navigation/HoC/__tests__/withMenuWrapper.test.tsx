
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import 'jest-styled-components';

import {
  withMenuWrapper,
  getJustifyContent,
  getFlexDirection,
  getBorderWidth,
  getOrder,
  getDisplay,
  getPadding,
} from '../withMenuWrapper';

import { withTestTheme } from '@relayr/ui-elements-themes';

enzyme.configure({ adapter: new Adapter() });

describe('ui-elements-navigation', () => {
  describe('HoC', () => {
    describe('withMenuWrapper', () => {
      it('Renders component with wrapper', () => {
        const WrappedComponent = () => <p />;
        const WithMenuWrapper = withMenuWrapper(WrappedComponent);
        const tree = TestRenderer.create(withTestTheme(
          <WithMenuWrapper
            display={[]}
            alignment="center"
            iconPosition="top"
          >
            Children
          </WithMenuWrapper>),
        )
          .toJSON();

        expect(tree).toMatchSnapshot();
      });

      it('getJustifyContent returns appropriate value', () => {
        expect(getJustifyContent('left')).toBe('flex-start');
        expect(getJustifyContent('right')).toBe('flex-end');
        expect(getJustifyContent('center')).toBe('center');
      });

      it('getFlexDirection returns appropriate value', () => {
        expect(getFlexDirection('left')).toBe('row');
        expect(getFlexDirection('right')).toBe('row');
        expect(getFlexDirection('top')).toBe('column');
      });

      it('getBorderWidth returns appropriate value', () => {
        expect(getBorderWidth('left')).toBe('0 0 0 4px');
        expect(getBorderWidth('right')).toBe('0 4px 0 0');
        expect(getBorderWidth('center')).toBe('0 0 0 4px');
      });

      it('getOrder returns appropriate value', () => {
        expect(getOrder('left')).toBe(-1);
        expect(getOrder('top')).toBe(-1);
        expect(getOrder('right')).toBe(1);
      });

      it('getDisplay returns appropriate value', () => {
        expect(getDisplay([], 'icon')).toBe('none');
        expect(getDisplay([], 'label')).toBe('none');
        expect(getDisplay(['icon'], 'icon')).toBe('block');
        expect(getDisplay(['label'], 'label')).toBe('block');
        expect(getDisplay(['label', 'icon'], 'icon')).toBe('block');
        expect(getDisplay(['label', 'icon'], 'label')).toBe('block');
      });

      it('getDisplay returns appropriate value', () => {
        expect(getPadding()).toBe('0 7px');
        expect(getPadding(true)).toBe('0 21px');
      });
    });
  });
});
