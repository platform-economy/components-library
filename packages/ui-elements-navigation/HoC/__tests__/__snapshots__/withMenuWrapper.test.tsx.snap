// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-navigation HoC withMenuWrapper Renders component with wrapper 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  background-color: #surface;
}

.c0 .c1 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-width: 0 0 0 4px;
}

.c0 .c2 {
  display: none;
  text-align: center;
}

.c0 .c3 {
  display: none;
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
  padding: 0 7px;
}

<div
  className="c0"
  display={Array []}
>
  <p />
</div>
`;
